File: src/RESR.Frontend/src/shared/types/comment.ts
import type { ResourceAuthor } from "@/shared/types/article";

export interface ResourceComment {
  idComment: number;
  content: string;
  createdAt: string;
  modifiedAt: string | null;
  deletedAt: string | null;
  idResource: number;
  idUser: number;
  author: ResourceAuthor;
  idParentComment: number | null;
}

export interface CreateResourceCommentRequest {
  content: string;
  idParentComment?: number | null;
}

File: src/RESR.Models/Comments/Comment.cs
namespace RESR.Models.Comments;

public sealed class Comment
{
    public int IdComment { get; set; }
    public string Content { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; }
    public DateTime? ModifiedAt { get; set; }
    public DateTime? DeletedAt { get; set; }
    public int IdResource { get; set; }
    public int IdUser { get; set; }
    public int? IdParentComment { get; set; }
}
